docarray/array/mixins/embed.py

```diff
@@ -22,18 +29,28 @@ def embed(
             `cpu` or `cuda`.
         :param batch_size: number of Documents in a batch for embedding
         :param to_numpy: if to store embeddings back to Document in ``numpy.ndarray`` or original framework format.
+        :param collate_fn: create a mini-batch of Input(s) from the given `DocumentArray`.  Default built-in collate_fn
+                is to use the `tensors` of the documents.
         :return: itself after modified.
         """
 
+        if collate_fn is None:
+
+            def default_collate_fn(da: 'DocumentArray'):
+                return da.tensors
+
+            collate_fn = default_collate_fn
+
         fm = get_framework(embed_model)
         getattr(self, f'_set_embeddings_{fm}')(
-            embed_model, device, batch_size, to_numpy
+            embed_model, collate_fn, device, batch_size, to_numpy
         )
         return self
 
     def _set_embeddings_keras(
         self: 'T',
         embed_model: 'AnyDNN',
+        collate_fn: 'CollateFnType',
```

Comment:
maybe we move all `collate_fn` to the last to be consistent? the reason is it's not frequently used as other arguments.

Reply:
in the `_set_embeddings_{fm}` function, the `collate_fn` is designed as positional parameter (as it is required always)